crates/datafusion_ext/src/system.rs

```diff
@@ -0,0 +1,21 @@
+use datafusion::error::Result as DataFusionResult;
+use datafusion::execution::TaskContext;
+use futures::Future;
+use std::pin::Pin;
+use std::sync::Arc;
+
+/// A system operation can execute an arbitrary operation.
+///
+/// This should be focused on operations that do not require user interactions
+/// (e.g. background operations like optimizing tables or running caching jobs).
+pub trait SystemOperation: Sync + Send {
```

Comment:
Would we ever have an unknown number of `SystemOperation`'s, or custom operations? Considering we are dealing with an exhaustive list of operations, I'm wondering if it'd make sense to use an enum here instead?

Reply:
Fair, will change